packages/contracts/contracts/attester-utils/ReimburseAttestation.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import {Unirep} from '../Unirep.sol';
+
+/**
+ * @title ReimburseAttestation
+ * @dev An attester can inherit this contract to hold budget and allow the callers of the contract (attester app users) to be reimbersed for attestations.
+ 	This would make the publicly funding operation of attesters more modular and transparent.
+	e.g. An attester could directly get funded from a gitcoin round for its operation.
+ */
+contract ReimburseAttestation {
+    bool public acceptDonations;
+    bool public acceptReReimburseUST;
+
+    mapping(address => bool) public whitelist;
+
+    Unirep public unirep;
+
+    // id of the attester this budget contract is deployed for
+    uint160 public attesterId;
+
+    enum TxType {
+        Signup,
+        UST
+    }
+    TxType public txType;
```

Comment:
It doesn't have to be a public variable. When you emit events, you just pass a `TxType.Signup` or `TxType. UST`.